packages/nodes-base/nodes/HackerNews/HackerNews.node.ts

```diff
@@ -0,0 +1,176 @@
+						value: 'article',
+						description: 'Returns a Hacker News article',
+					},
+					{
+						name: 'Username',
+						value: 'username',
+						description: 'Returns a Hacker News user',
+					},
+					{
+						name: 'Query',
+						value: 'query',
+						description: 'Returns articles based on query',
+					}
+				],
+				default: 'article',
+				description: 'The operation to perform',
+			},
+			// ----------------------------------
+			//         Article
+			// ----------------------------------
+			{
+				displayName: 'Article Id',
+				name: 'id',
+				type: 'string',
+				required: true,
+				default: '',
+				description: 'The id of the article to be returned',
+				displayOptions: { show: { operation: [ 'article' ] } },
+			},
+			// ----------------------------------
+			//         Username
+			// ----------------------------------
+			{
+				displayName: 'Username',
+				name: 'username',
+				type: 'string',
+				required: true,
+				default: '',
+				description: 'The user to be returned',
+				displayOptions: { show: { operation: [ 'username' ] } },
+			},
+			// ----------------------------------
+			//         Query keyword
+			// ----------------------------------
+			{
+				displayName: 'Keyword',
+				name: 'keyword',
+				type: 'string',
+				default: '',
+				displayOptions: { show: { operation: [ 'query' ] } },
+				description: 'The keyword for filtering the results of the query.',
+			},
+			// ----------------------------------
+			//         Query tag
+			// ----------------------------------
+			{
+				displayName: 'Tags',
+				name: 'tags',
+				type: 'multiOptions', // undocumented
+				options: [
+					{
+						name: 'Story',
+						value: 'story',
+						description: 'Returns query results filtered by story tag',
+					},
+					{
+						name: 'Comment',
+						value: 'comment',
+						description: 'Returns query results filtered by comment tag',
+					},
+					{
+						name: 'Poll',
+						value: 'poll',
+						description: 'Returns query results filtered by poll tag',
+					},
+					{
+						name: 'Show HN',
+						value: 'show_hn', // snake case per HN tags
+						description: 'Returns query results filtered by Show HN tag',
+					},
+					{
+						name: 'Ask HN',
+						value: 'ask_hn', // snake case per HN tags
+						description: 'Returns query results filtered by Ask HN tag',
+					},
+					{
+						name: 'Front Page',
+						value: 'front_page', // snake case per HN tags
+						description: 'Returns query results filtered by Front Page tag',
+					}
+				],
+				default: '',
+				displayOptions: { show: { operation: [ 'query' ] } },
+				description: 'Tag for filtering the results of the query.',
+			},
+		]
+	};
+
+
+	async execute(this: IExecuteFunctions): Promise<INodeExecutionData[][]> {
+		const items = this.getInputData();
+		const returnData: IDataObject[] = [];
+
+		const operation = this.getNodeParameter('operation', 0) as string;
+
+		for (let i = 0; i < items.length; i++) {
```

Comment:
Why do you use a loop here? What is the case in which `i` is not 0? I might totally be mistaken on how this is working, but just a question.

Reply:
@erin2722 If the node input has 20 items then `i` will take from 0 to 19.